### PR TITLE
feat: add other policies in CEL expressions - Part 7

### PR DESCRIPTION
**Pull request by Chandan-DK**
## Related Issue(s)
Partially addresses https://github.com/kyverno/policies/issues/891

## Description
This PR includes the conversion of policies in the other folder to Kyverno CEL policies. Conversion of the policies is done in multiple PRs. 

**Note:** Kyverno CLI tests cannot be added for some of these policies as of now because some features need to be added:
1. Support for parameter resources in Kyverno CLI tests [#9431](https://github.com/kyverno/kyverno/issues/9431)

**Policies converted in this PR:**

- [x] advanced-restrict-image-registries

- [x] deny-commands-in-exec-probe
- [x] exclude-namespaces-dynamically
- [x] limit-hostpath-vols
- [x] require-container-port-names
- [x] require-emptydir-requests-limits
- [x] require-non-root-groups
- [x] require-qos-guaranteed
- [x] restrict-edit-for-endpoints
- [x] restrict-jobs
- [x] restrict-node-label-creation

- [x] restrict-pod-controller-serviceaccount-updates
- [x] topologyspreadconstraints-policy

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: other-cel/block-pod-exec-by-namespace/block-pod-exec-by-namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      any:
+      - resources:
+          kinds:
+          - Pod/exec
+          operations:
+          - CONNECT
+    validate:
+      cel:
+        expressions:
+          - expression: "request.namespace != 'pci'"
```
Comment (MariamFahmy98) [inline other-cel/block-pod-exec-by-namespace/block-pod-exec-by-namespace.yaml:30]: Can we use the `namespaceObject` instead?
```
expression: "namespaceObject.metadata.name != 'pci'"
```

**Review by MariamFahmy98 — COMMENTED** [review]
Could you please fix the merge conflicts?

File: other-cel/block-pod-exec-by-namespace/block-pod-exec-by-namespace.yaml (hunk 1 above)
Comment (Chandan-DK) [reply other-cel/block-pod-exec-by-namespace/block-pod-exec-by-namespace.yaml]: Sure. I've made the changes.

**Comment by chipzoller** [comment]
Ping @MariamFahmy98 to help either get the outstanding PRs by @Chandan-DK either merged or closed. These have been out there for some time now.

**Review by MariamFahmy98 — APPROVED** [review]

**Review by MariamFahmy98 — APPROVED** [review]

**Pull request merged**